From: Ayoyimika Ajibade <ayoyimikaajibade@gmail.com>
Date: Sun, 18 Aug 2024 20:13:35 +0000
Subject: move to @rollup/plugin-typescript

Forwarded: not-needed
Last-Update: 2021-11-04

move from @wessberg/rollup-plugin-ts to @rollup/plugin-typescript
---
 package.json     | 2 +-
 rollup.config.js | 5 +++--
 tsconfig.json    | 8 +++-----
 3 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/package.json b/package.json
index 3426665..3e2721e 100644
--- a/package.json
+++ b/package.json
@@ -57,7 +57,7 @@
     "@types/node": "^14.0.27",
     "@typescript-eslint/eslint-plugin": "^4.0.0",
     "@typescript-eslint/parser": "^4.0.0",
-    "@wessberg/rollup-plugin-ts": "^1.3.2",
+    "@rollup/plugin-typescript": "^6.0.0",
     "c8": "^7.3.0",
     "chai": "^4.2.0",
     "chalk": "^4.1.0",
diff --git a/rollup.config.js b/rollup.config.js
index ec8a2ac..19c70c6 100644
--- a/rollup.config.js
+++ b/rollup.config.js
@@ -1,8 +1,9 @@
-import ts from '@wessberg/rollup-plugin-ts'
+import ts from '@rollup/plugin-typescript';
 
 const output = {
+  dir: "build",
   format: 'cjs',
-  file: './build/index.cjs',
+  entryFileNames: 'index.cjs',
   exports: 'default'
 }
 
diff --git a/tsconfig.json b/tsconfig.json
index d5d8c21..ea27d3e 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -4,14 +4,12 @@
     "outDir": "build",
     "rootDir": ".",
     "sourceMap": false,
+    "lib": ["es2019"],
     "target": "es2017",
     "moduleResolution": "node",
     "module": "es2015"
-  },  
+  },
   "include": [
     "lib/**/*.ts"
-  ],
-  "exclude": [
-    "lib/cjs.ts"
   ]
-}
\ No newline at end of file
+}
